frontend/src/components/query-component/query-component.tsx

```diff
@@ -5,17 +5,20 @@ import { Send } from '@mui/icons-material'
 import { useCallback } from 'react'
 
 type QueryComponentProps = {
-  sendQuery: (query: string) => void
+  sendQuery: (query: string, setLoading: (loading: boolean) => void) => void
+  setLoading: (loading: boolean) => void
+  loading: boolean
 }
-export function QueryComponent({ sendQuery }: QueryComponentProps) {
+export function QueryComponent({ sendQuery, setLoading, loading }: QueryComponentProps) {
```

Comment:
`loading` state is never used, maybe we do not need to pass that prop to this component?

Reply:
Removed accordingly in commit [22da40b](https://github.com/nknguyenhc/luminous-zephyr/pull/16/commits/22da40b7407cdec01ac9a4f0aef421d28bec604d)